### DISCUSSION TITLE
Memory Consumption (RAM) in fast dds

### DISCUSSION THREAD

**Discussion by AnoopEmbeddedEngineer** [eProsima/Fast-DDS#5434] [General]
When i tried running a sample ros2 application.. why the application is consuming more memory (~1 gb) for subscribing...

I'm experiencing high memory consumption (~1 GB) in my ROS2 application on the subscriber side, even when sending smaller data sizes. Here are the details:

Publisher Command: ./publisher_node_v4h2 -sp1 /dev/zero -ds1 10485 -f1 1 -t1 anoop -q1 1
Subscriber Command: ./subscriber_node_v4h2 -dp1 /dev/null -t1 anoop
Questions:

What are the typical memory allocations taking place in RAM for ROS2 nodes, publishers, and subscribers?
Are there any QoS settings that can help reduce memory usage?
Any tips on monitoring and optimizing memory usage in ROS2 applications?

---------------------------Additional data ------------------------------------------------------------
root@v4x:~# 
root@v4x:~# ros2 topic info /anoop --verbose
Type: std_msgs/msg/ByteMultiArray

Publisher count: 1

Node name: publisher_node
Node namespace: /
Topic type: std_msgs/msg/ByteMultiArray
Endpoint type: PUBLISHER
GID: 01.0f.9d.ac.3e.03.71.e3.01.00.00.00.00.00.12.03.00.00.00.00.00.00.00.00
QoS profile:
  Reliability: RELIABLE
  History (Depth): UNKNOWN
  Durability: VOLATILE
  Lifespan: Infinite
  Deadline: Infinite
  Liveliness: AUTOMATIC
  Liveliness lease duration: Infinite

Subscription count: 1

Node name: subscriber_node
Node namespace: /
Topic type: std_msgs/msg/ByteMultiArray
Endpoint type: SUBSCRIPTION
GID: 01.0f.b0.5a.f8.02.60.d4.01.00.00.00.00.00.12.04.00.00.00.00.00.00.00.00
QoS profile:
  Reliability: RELIABLE
  History (Depth): UNKNOWN
  Durability: VOLATILE
  Lifespan: Infinite
  Deadline: Infinite
  Liveliness: AUTOMATIC
  Liveliness lease duration: Infinite

root@v4x:~#

**Comment by paxifaer** [comment]
Maybe it is related to your data type.can you show me your idl?

**Comment by AnoopEmbeddedEngineer** [comment]
Type ByteMultiArray is used


--------------------------------------------------------------------------------------------------------------------------------------------------------

// generated from rosidl_adapter/resource/msg.idl.em
// with input from std_msgs/msg/ByteMultiArray.msg
// generated code does not contain a copyright notice

#include "std_msgs/msg/MultiArrayLayout.idl"

module std_msgs {
  module msg {
    @verbatim (language="comment", text=
      "This was originally provided as an example message." "\n"
      "It is deprecated as of Foxy" "\n"
      "It is recommended to create your own semantically meaningful message." "\n"
      "However if you would like to continue using this please use the equivalent in example_msgs.")
    struct ByteMultiArray {
      @verbatim (language="comment", text=
        "Please look at the MultiArrayLayout message definition for" "\n"
        "documentation on all multiarrays." "\n"
        "specification of data layout")
      std_msgs::msg::MultiArrayLayout layout;

      @verbatim (language="comment", text=
        "array of data")
      sequence<octet> data;
    };
  };
};

**Reply by paxifaer** [reply]
I think this issue does not caused by fastdds,maybe your program has a memory leak,You can check it by using valgrind.
